h2o-py/h2o/frame.py

```diff
@@ -146,6 +146,12 @@ def _upload_python_object(self, python_obj, destination_frame=None, header=0, se
         if not column_names:
             column_names = col_header
 
+        if skipped_columns:
+            column_names = [column_name for column_name in column_names[:len(column_names)-len(skipped_columns)]]
```

Comment:
This ignores the skipped columns so in some cases the skipped column is present in the resulting frame.

Let's have a frame:
```
| a | b | c |
+---+---+---+
| 1 | 2 | 3 |
| 4 | 5 | 6 |
```

and if we have `skipped_columns=[0,1]`, then we would get just:
```
| a |
+---+
| 1 | 
| 4 |
```

but I would expect to get:

```
| c |
+---+
| 3 |
| 6 |
```

Also note that it's likely that `skipped_columns` should support inputs  that are column indices (`List[int]`) or column names (`List[str]`).

Reply:
@tomasfryda I understand your point, nevertheless i understand the problem and will try to solve it.

Are there any other cases I should be aware of?

Reply:
You should also make sure you modify the [csv writing](https://github.com/h2oai/h2o-3/pull/16071/files#diff-85c7bce9c6153a47d6a744ec3d597218e0366b8432642e09fc12bb27536c5bf8R156-R169) accordingly,  so that only those columns that are not skipped are written. And please write a test for those cases when there's `skipped_columns` are at the beginning, the end, and in the middle (and possibly more cases depending on your actual implementation).

And while you're at it, you can also improve the code so that it makes sure that the `tmp_path` is removed, e.g.:

```python
tmp_handle, tmp_path = tempfile.mkstemp(suffix=".csv")
try:
    .....
finally:
    os.remove(tmp_path)
```

Reply:
Got it, also earlier i ran the command `python3 -m unittest pyunit_h2oH2OFrame.py` to run the tests and it worked, but this time it is unable to start the local h2o server and is unable to connect to one. Is there any other way the tests are run for h2o?

Reply:
> Got it, also earlier i ran the command `python3 -m unittest pyunit_h2oH2OFrame.py` to run the tests and it worked, but this time it is unable to start the local h2o server and is unable to connect to one. Is there any other way the tests are run for h2o?

Reply:
@tomasfryda I made more changes based on the requirements and the original issue description. Also I think there is no need to make any in [csv writing](https://github.com/h2oai/h2o-3/pull/16071/files#diff-85c7bce9c6153a47d6a744ec3d597218e0366b8432642e09fc12bb27536c5bf8R156-R169) because it takes the updated `column_names` as argument.

Please correct me if I'm wrong